msticpy/data/drivers/github_data_types.py

```diff
@@ -0,0 +1,288 @@
+            formatted_qname = format_query_name(cur_query.name)
+            sentinel_features = list(attr.asdict(cur_query).keys())
+            sentinel_features.remove('description')
+            sentinel_features.remove('query')
+            dict_to_write['sources'][formatted_qname] = dict()
+            dict_to_write['sources'][formatted_qname]['description'] = cur_query.description
+            dict_to_write['sources'][formatted_qname]['metadata'] = {"pivot": {"short_name": "interface", "direct_func_entities": ["Host"]}, "sentinel": dict()}
+            cur_query_dict = attr.asdict(cur_query)
+            for feature in sentinel_features:
+                dict_to_write['sources'][formatted_qname]['metadata']['sentinel'][feature] = cur_query_dict[feature]
+#             dict_to_write['sources'][format_query_name(cur_query.name)] = attr.asdict(cur_query)
+            dict_to_write['sources'][formatted_qname]['args'] = dict()
+            dict_to_write['sources'][formatted_qname]['args']['query']= cur_query.query
+            if qtype == "detection":
+                dict_to_write['sources'][formatted_qname]['parameters'] = dict()
+            elif qtype == "hunting":
+                dict_to_write['sources'][formatted_qname]['parameters'] = {"start": {"description": "Query start time", "type": "datetime", "aliases": ["StartTimeISO"]}, "end": {"description": "Query end time", "type": "datetime", "aliases": ["EndTimeISO"]}}
+        cur_index += ct_folders[f]
+#         ret_dicts.append(dict_to_write)
+        
+        try:
+            query_text = yaml.safe_dump(dict_to_write, encoding="utf-8")
+        except yaml.YAMLError as e:
+            print(e)
+        
+        try:
+            path_main = os.path.join(def_path, "saved_queries/")
+            path_type = os.path.join("saved_queries/" + qtype)
+            if not os.path.exists(path_main):
+                path = os.path.join(def_path, "saved_queries")
+                os.mkdir(path)
+            if not os.path.exists(path_type):
+                path = os.path.join("saved_queries", qtype)
+                os.mkdir(path)
+            Path("saved_queries/" + qtype + "/" + f + ".yaml").write_text(str(query_text))
+        except OSError as e:
+            print(e)
+    print("done writing query files")
+
+
+def import_parse_write_queries():
+    print('downloading files from github')
+    def_path = Path.joinpath(Path(os.getcwd()))
+    # Download the Microsoft Sentinel Github repo as ZIP
+    azsentinel_git_url = 'https://github.com/Azure/Azure-Sentinel/archive/master.zip'
+    get_sentinel_queries_from_github(git_url=azsentinel_git_url, outputdir=def_path)
+    base_dir = str(def_path) + "/Azure-Sentinel-master"
+    
+    print('processing queries')
+    detection_queries = parse_yaml(parent_dir=base_dir, child_dir="Detections")
+    hunting_queries = parse_yaml(parent_dir=base_dir, child_dir="Hunting Queries")
+
+    d_child_dir = "Detections"
```

Comment:
Shouldn't need these sections I think if you're extracting and saving the folder name when reading the yaml files.